docs/compat.rst

```diff
@@ -11,6 +11,13 @@ Backwards Compatibility
 
 .. currentmodule:: gpiozero
 
+We aim to ensure that gpiozero will run on all versions of python included in releases
```

Comment:
I wonder if it might make sense to split this Docker / devcontainer stuff out into a separate PR, so that it can be reviewed and merged independently of this one?